### PR TITLE
schemaTracker: fix operate-schema --flush panic

### LINKED ISSUE #1724: `operate-schema set` will use `--flush` as default action
## Feature Request

**Is your feature request related to a problem? Please describe:**

If we not `--flush`, which means write a table checkpoint, and following error happens, will cause "drop table in schema tracker because no checkpoint exists" so the `operate-schema set` is silently disappeared.

**Describe the feature you'd like:**

as title

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
close #1724 #1730

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

**Comment by lance6716** [comment]
```
[2021-06-22T07:09:57.252Z] command: operate-schema get -s mysql-replica-01,mysql-replica-02 sequence_sharding_optimistic -d sharding_seq_opt -t t2 "result": true count: 0 != expected: 3

[2021-06-22T07:09:57.252Z] fail to run: [operate-schema get -s mysql-replica-01,mysql-replica-02 sequence_sharding_optimistic -d sharding_seq_opt -t t2]

[2021-06-22T07:09:57.252Z] Error: --flush flag is only used to set schema
```

File: syncer/checkpoint.go [hunk 1]
```diff
@@ -563,7 +563,10 @@ func (cp *RemoteCheckPoint) FlushPointWithTableInfo(tctx *tcontext.Context, sour
 	sqls := make([]string, 0, 1)
 	args := make([][]interface{}, 0, 10)
 
-	point := cp.points[sourceSchema][sourceTable]
+	point, ok := cp.points[sourceSchema][sourceTable]
```
Comment (lance6716) [inline syncer/checkpoint.go:566]: will `cp.points[sourceSchema]` returns `nil` 🤔

File: tests/downstream_more_column/conf/diff_config.toml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+host = "127.0.0.1"
+port = 3306
+user = "root"
+password = "123456"
+instance-id = "source-1"
+
+[target-db]
+host = "127.0.0.1"
+port = 4000
+user = "test"
```
Comment (lance6716) [inline tests/downstream_more_column/conf/diff_config.toml:39]: a new user 🤔

File: tests/downstream_more_column/run.sh [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+tb="t1"
+
+function get_master_status() {
+	arr=$(echo "show master status;" | MYSQL_PWD=123456 mysql -uroot -h127.0.0.1 -P3306 | awk 'NR==2')
+	echo $arr
+}
+
+function run() {
+	run_sql_file $cur/data/db1.prepare.sql $MYSQL_HOST1 $MYSQL_PORT1 $MYSQL_PASSWORD1
+	# create table in tidb with auto-random
```
Comment (lance6716) [inline tests/downstream_more_column/run.sh:18]:
```suggestion
	# create table in tidb with AUTO_INCREMENT
```

File: tests/downstream_more_column/run.sh [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	# check incremental data
+	run_sql_tidb_with_retry "select count(1) from ${db}.${tb} where c1>100 and c1<1000;" "count(1): 2"
+
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"stop-task test" \
+		"\"result\": true" 2
+
+	# start DM task in incremental mode
+	# schemaTracker create table from downstream
+	master_status=($(get_master_status))
```
Comment (lance6716) [inline tests/downstream_more_column/run.sh:47]: confused with this line, will test later

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: tests/downstream_more_column/run.sh [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+set -eu
+
+cur=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+source $cur/../_utils/test_prepare
+WORK_DIR=$TEST_DIR/$TEST_NAME
+db="downstream_more_column"
+tb="t1"
+
+function get_master_status() {
```
Comment (lance6716) [inline tests/downstream_more_column/run.sh:11]: maybe we could use get_master_status from upstream_switch/lib.sh

Comment (GMHDBJD) [reply tests/downstream_more_column/run.sh]: Add in test_prepare so we can reuse it.

**Review by lance6716 — APPROVED** [review]

File: tests/_utils/test_prepare [hunk 6]
```diff
@@ -276,4 +276,9 @@ function init_cluster(){
     sed -i "/relay-binlog-name/i\relay-dir: $WORK_DIR/worker2/relay_log" $WORK_DIR/source2.yaml
     dmctl_operate_source create $WORK_DIR/source1.yaml $SOURCE_ID1
     dmctl_operate_source create $WORK_DIR/source2.yaml $SOURCE_ID2
-}
\ No newline at end of file
+}
+
+function get_master_status() {
```
Comment (Ehco1996) [inline tests/_utils/test_prepare:281]: there is a same func in `tests/upsteam_switch/lib.sh`, do we need to delete that one?

Comment (GMHDBJD) [reply tests/_utils/test_prepare:281]: Seems some variables in test_prepare need set in github action again, I keep two version of this function now.

**Comment by GMHDBJD** [comment]
cc @Ehco1996

**Comment by Ehco1996** [comment]
LGTM

**Pull request merged**